contracts/test/CW721ERC721WraperTest.t.sol

```diff
@@ -0,0 +1,228 @@
+
+address constant MockCallerEVMAddr = 0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266;
+address constant MockOperatorEVMAddr = 0xF39fD6e51Aad88F6f4CE6AB8827279CFffb92267;
+string constant MockCallerSeiAddr = "sei19zhelek4q5lt4zam8mcarmgv92vzgqd3ux32jw";
+string constant MockOperatorSeiAddr = "sei1vldxw5dy5k68hqr4d744rpg9w8cqs54x4asdqe";
+string constant MockCWContractAddress = "sei14hj2tavq8fpesdwxxcu44rty3hh90vhujrvcmstl4zr3txmfvw9sh9m79m";
+
+contract MockWasmd is IWasmd {
+
+    // Transactions
+    function instantiate(
+        uint64,
+        string memory,
+        bytes memory,
+        string memory,
+        bytes memory
+    ) external pure returns (string memory, bytes memory) {
+        return (MockCWContractAddress, bytes(""));
+    }
+
+    function execute(
+        string memory contractAddress,
+        bytes memory,
+        bytes memory
+    ) external pure returns (bytes memory) {
+        require(keccak256(abi.encodePacked(contractAddress)) == keccak256(abi.encodePacked(MockCWContractAddress)), "wrong CW contract address");
+        return bytes("");
+    }
+
+    // Queries
+    function query(string memory, bytes memory) external pure returns (bytes memory) {
+        return bytes("");
+    }
+}
+
+contract MockJson is IJson {
+    function extractAsBytes(bytes memory, string memory) external pure returns (bytes memory) {
+        return bytes("extracted bytes");
+    }
+
+    function extractAsBytesList(bytes memory, string memory) external pure returns (bytes[] memory) {
+        return new bytes[](0);
+    }
+}
+
+contract MockAddr is IAddr {
+    function getSeiAddr(address addr) external pure returns (string memory) {
+        if (addr == MockCallerEVMAddr) {
+            return MockCallerSeiAddr;
+        }
+        return MockOperatorSeiAddr;
+    }
+
+    function getEvmAddr(string memory addr) external pure returns (address) {
+        if (keccak256(abi.encodePacked(addr)) == keccak256(abi.encodePacked(MockCallerSeiAddr))) {
+            return MockCallerEVMAddr;
+        }
+        return MockOperatorEVMAddr;
+    }
+}
+
+contract CW721ERC721WrapperTest is Test {
+    event Transfer(address indexed from, address indexed to, uint256 indexed tokenId);
+    event Approval(address indexed owner, address indexed approved, uint256 indexed tokenId);
+    event ApprovalForAll(address indexed owner, address indexed operator, bool approved);
+
+    CW721ERC721Wrapper wrapper;
+    MockWasmd mockWasmd;
+    MockJson mockJson;
+    MockAddr mockAddr;
+
+    function setUp() public {
+        wrapper = new CW721ERC721Wrapper(MockCWContractAddress, "name", "symbol");
+        mockWasmd = new MockWasmd();
+        mockJson = new MockJson();
+        mockAddr = new MockAddr();
+        vm.etch(WASMD_PRECOMPILE_ADDRESS, address(mockWasmd).code);
+        vm.etch(JSON_PRECOMPILE_ADDRESS, address(mockJson).code);
+        vm.etch(ADDR_PRECOMPILE_ADDRESS, address(mockAddr).code);
+    }
+
```

Comment:
might be good to test all functions including things like name and symbol and decimals just to make sure they're set properly.

Reply:
good point, added